homeassistant/components/template/binary_sensor.py

```diff
@@ -59,14 +63,22 @@
         entity_picture_template = device_config.get(CONF_ENTITY_PICTURE_TEMPLATE)
         entity_ids = set()
         manual_entity_ids = device_config.get(ATTR_ENTITY_ID)
+        attribute_templates = device_config.get(CONF_ATTRIBUTE_TEMPLATES)
 
         invalid_templates = []
 
-        for tpl_name, template in (
-            (CONF_VALUE_TEMPLATE, value_template),
-            (CONF_ICON_TEMPLATE, icon_template),
-            (CONF_ENTITY_PICTURE_TEMPLATE, entity_picture_template),
-        ):
+        templates = dict(
+            (
+                (CONF_VALUE_TEMPLATE, value_template),
+                (CONF_ICON_TEMPLATE, icon_template),
+                (CONF_ENTITY_PICTURE_TEMPLATE, entity_picture_template),
+            )
+        )
+
+        if attribute_templates is not None:
+            templates.update(attribute_templates)
```

Comment:
This is not good, it means attribute templates could override the other 3 templates.